### ISSUE TITLE
Navigation on coastal flowlines is unconstrained

### ISSUE THREAD

**Issue by pauldzy** [ACWI-SSWD/nldi-services#39]
This issue is a followup to the working group discussion back in October.

NHDPlus Navigation is constrained by cost (distance or time) taken from the flowline VAA attributes.  This logic dates to NHDPlus 1.0 days when all flowline VAA attributes had some kind of cost value.  

However when NHDPlus 2.0 was released, it included for the first time coastal flowlines - all of which have a distance cost of zero.  The NHDPlus team has verified this is as intended.

The problem occurs when for whatever reason a coastal flowline is selected for navigation.  For example a user submits a simple request to navigation "upstream" for 5 KM from coast flowline ComID 8379397 near Lusby on the Maryland coast of the Chesapeake Bay.
![capture1](https://cloud.githubusercontent.com/assets/4131067/20312965/5c5e6fba-ab23-11e6-9d56-540bf84d4509.PNG)

As coastal flowlines have a distance cost of zero the navigation will run the entire linked distance to the Mexican border
![capture](https://cloud.githubusercontent.com/assets/4131067/20313013/84ef393c-ab23-11e6-944d-b26b0e5c118a.PNG)

As this result is probably undesired and as this result is onerous to the server and client, it might be best to disallow navigation from and upon coastal flowlines.

**Comment by pauldzy** [comment]
At the November meeting Tommy made the comment that coastal navigation can be of use to beach indexing purposes.  This is quite true.  I would rephrase my recommendation avoid to say that cost-based (upstream downstream) navigation from coastal flowlines should be disallowed but point to point navigation on coastal flowlines is fine.

**Comment by kkehl-usgs** [comment]
How can we determine what is a coastal flowline and what isn't?  Is it a numbering convention in the comId?

**Comment by pauldzy** [comment]
Coastal flowlines have an FCode value of 56600 that identifies them.  While point-to-point coastal flowline navigation has some utility for the EPA Beach program, overall its not really that useful as few users see coastlines as entities that flow.  In version 3 of EPA navigation we just exclude coastal flowlines entirely from the navigation set and thus that solves the problem.  

https://github.com/pauldzy/NHDPlus_Navigation_PG/blob/3c73715d37093026d4f41f3e1135abdc580dffef/src/nhdplus_navigation30/MaterializedViews/PLUSFLOWLINEVAA_NAV.sql#L203

**Comment by dblodgett-usgs** [comment]
My vote is to remove coastal flowlines -- at least for the time being. May work over in the nhdplusTools R package also disallows coastal flowlines. (I remove Coastline features here: https://github.com/usgs-r/nhdplusTools/blob/master/R/prep_nhdplus.R#L65)

If we want to bring them back at some point, it should be done with a little different connection to the API.

**Comment by dblodgett-usgs** [comment]
We should get this fixed @brownd-usgs -- This query: https://labs.waterdata.usgs.gov/api/nldi/linked-data/wqp/USGS-450608087074101/navigation/UM/flowlines?distance=1&f=json 

Is a WQP site that causes the issue. 

https://www.waterqualitydata.us/provider/NWIS/USGS-WI/USGS-450608087074101/

We need to avoid coastal flowlines in navigation. (FCode value of 56600)
